### PR TITLE
Add DeepFinance sample

### PR DESCRIPTION
**Pull request by TaoShuchang**
## Summary

This PR adds **DeepFinance**, a comprehensive financial research agent training framework built on AgentScope Tuner. The system uses GRPO (Group Relative Policy Optimization) reinforcement learning to train LLMs for autonomous financial research report generation.

### Core Architecture
- **DeepFinance Judge Engine**: Multi-dimensional scoring system for evaluating research reports
- **Workflow Design**: Two-stage deep research process (outline → investigation → report)
- **Training Pipeline**: Distributed rollout, judge, and trainer components via Ray

### Multi-dimensional Evaluation System

| Dimension | Grader | Purpose |
|-----------|--------|---------|
| Analysis Sufficiency | RM Gallery | Overall report quality |
| Presentation Quality | PresentationQuality | Report formatting and structure |
| Citation Norm | Grounding | Citation coverage and authenticity |
| Citation Logic Audit | Audit | Logical entailment of citations |

### Components Added
- `tuner/deep_finance/judge/`: Judge graders (audit, finance, grounding, presentation_quality)
- `tuner/deep_finance/metric_helper/`: Reward and tool metric utilities
- `tuner/deep_finance/prompt/`: Financial analyst prompts and tool builders
- Training scripts: `deepfinance_tuner.sh` and `deepfinance_tuner_single.sh`

### Documentation
- README with Quick Start guide
- Blog articles (Chinese and English versions)
- Configuration templates

## Commits in this PR
- feat(deepfinance): add DeepFinance scoring engine, data preparation, and example protocol
- feat(judge): add audit, EBTU traceability and finance composition graders
- feat(metric): add reward and tool metric helper utilities
- refactor(judge): improve grounding and presentation quality graders
- refactor(core): redesign judge and workflow architecture
- perf(deep_finance): optimize training settings
- refactor(deep_finance): add article saving mechanism
- feat(deepfinance): add blog and readme
- chore(deep_finance): improve environment setup and training scripts
- fix(tuner): reduce parallel runners and extend timeout for deepfinance

## Files Changed
- 35 files changed
- 5,458 insertions(+), 8 deletions(-)

### CODE REVIEW

File: tuner/deep_finance/README.md
Comment (pan-x-c) [inline tuner/deep_finance/README.md:1]: Please add an English version README

File: tuner/deep_finance/blog_cn.md [hunk 1]
```diff
@@ -0,0 +1,531 @@
+# How to Train a Finance Deep Research Agent: Design Principles and Lessons Learned
```
Comment (pan-x-c) [inline tuner/deep_finance/blog_cn.md]: Could it be merged into README?

Comment (TaoShuchang) [reply tuner/deep_finance/blog_cn.md]: I have deleted the blog.

File: tuner/deep_finance/README.md
Comment (TaoShuchang) [reply tuner/deep_finance/README.md:1]: I have added a english version of README

**Review by pan-x-c — APPROVED** [review]

**Review by rayrayraykk — APPROVED** [review]
